internal/propelauth/client.go

```diff
@@ -19,6 +21,23 @@ type PropelAuthClient struct {
 	ApiKey     string
 }
 
+type PropelAuthApiError struct {
+	ErrorCode string `json:"error_code"`
+	UserFacingError string `json:"user_facing_error"`
+}
+
+func ConvertStringErrorToPropelAuthError (err error) (*PropelAuthApiError, error) {
```

Comment:
Maybe we make this private and only use it within the `propelauth` package and return back error codes for the error string where relevant. Then we don't have to worry about string parsing on line 33 (not so bad but would break if anyone changed "error on response: " upstream anywhere)

Reply:
hows the newest change?

Reply:
looks great!